File: Updog.Client/src/core/pagination/pagination-info.ts
/**
 * Information about a paged result set from the backend.
 */
export class PaginationInfo {
    /**
     * Create a new set of pagination information.
     * @param pageNumber The current page number.
     * @param pageSize The size of the page.
     * @param totalCount The total record count available.
     */
    constructor(public pageNumber: number, public pageSize: number, public totalCount: number) {}

    /**
     * Check to see if the paged result set has another one that can be
     * retrieved.
     */
    public hasNextPage(): boolean {
        return this.totalCount > this.pageSize * (this.pageNumber + 1);
    }

    /**
     * Check to see if the paged result set has a page previous to this one.
     */
    public hasPreviousPage(): boolean {
        return this.pageNumber > 0;
    }

    public pageCount(): number {
        return this.totalCount / this.pageSize;
    }
}

File: Updog.Domain/Core/Paging/PaginationInfo.cs
using System;

namespace Updog.Domain.Paging {
    /// <summary>
    /// Paging info.
    /// </summary>
    public sealed class PaginationInfo {
        #region Properties
        /// <summary>
        /// The current page number being retrieved.
        /// </summary>
        public int PageNumber { get; }

        /// <summary>
        /// The desired page size. (May be smaller if not enough resources
        /// are available to fill all slots)
        /// </summary>
        public int PageSize { get; }

        /// <summary>
        /// The total number of pages available based on the current page size.
        /// </summary>
        public int TotalRecordCount { get; }

        /// <summary>
        /// The offset of the first record.
        /// </summary>
        public int Offset => PageNumber * PageSize;
        #endregion

        #region Constructor(s)
        /// <summary>
        /// Create a new set of pagination info.
        /// </summary>
        /// <param name="pageNumber">The current page number.</param>
        /// <param name="pageSize">The size of the page.</param>
        /// <param name="totalRecordCount">Total available pages.</param>
        public PaginationInfo(int pageNumber, int pageSize, int totalRecordCount = 0) {
            PageNumber = pageNumber;
            PageSize = pageSize;
            TotalRecordCount = totalRecordCount;
        }
        #endregion
    }
}